airbyte-ci/connectors/pipelines/pipelines/sentry_utils.py

```diff
@@ -0,0 +1,40 @@
+import sentry_sdk
+from connector_ops.utils import Connector
+
+
+def with_step_context(func):
+    def wrapper(self, *args, **kwargs):
+        with sentry_sdk.configure_scope() as scope:
+            step_name = self.__class__.__name__
+            scope.set_tag("pipeline_step", step_name)
+            scope.set_context("Pipeline Step", {
+                'name': step_name,
+                'step_title': self.title,
+                'max_retries': self.max_retries,
+                'retry_count': self.retry_count,
+            })
+
+            if hasattr(self.context, "connector"):
+                connector: Connector = self.context.connector
+                scope.set_tag("connector", connector.technical_name)
+                scope.set_context("Connector", {
+                    'name': connector.name,
+                    'technical_name': connector.technical_name,
+                    'language': connector.language,
+                })
+
+            return func(self, *args, **kwargs)
+    return wrapper
+    
+
+def with_command_context(func):
+    def wrapper(self, ctx, *args, **kwargs):
+        with sentry_sdk.configure_scope() as scope:
+            scope.set_tag("pipeline_command", self.name)
+            scope.set_context("Pipeline Command", {
+                'name': self.name,
+                'params': self.params,
+            })
+            scope.set_context("Click Context", ctx.obj)
```

Comment:
The ctx.obj has secrets like SA account, GitHub PAT etc., will they be obfuscated by Sentry?

Reply:
Yep! 
<img width="351" alt="image" src="https://github.com/airbytehq/airbyte/assets/4368928/5ac263a1-2d3e-44b5-a61e-432f74c05d8b">